.github/workflows/rust.yml

```diff
@@ -248,3 +249,34 @@ jobs:
     steps:
       - uses: actions/checkout@v3
       - uses: EmbarkStudios/cargo-deny-action@v1
+
+  move-prover:
+    name: move-prover
+    needs: diff
+    if: needs.diff.outputs.isMove == 'true'
+    runs-on: [ubuntu-ghcloud]
```

Comment:
Could we please add a timeout just in case, say 20 minutes?

Reply:
20 minutes? Seems like a long time for a required check, right?

Reply:
Average running time (https://github.com/MystenLabs/sui/actions/workflows/move-prover.yml) is about 7min and the longest I see there is 13min, so I figured 20min timeout would be useful just as fail-safe.

Reply:
Ok, added a 20 minute wait.